### ISSUE TITLE
Adjoint differentiation method fails with jax interface.

### ISSUE THREAD

**Issue by albi3ro** [PennyLaneAI/pennylane#1345]
With the jax interface, the `adjoint` differentiation method produces incorrect results.  The jax interface does not have tests for adjoint diff as well; hence this has fallen through the cracks. The jax interface still works fine with other differentiation methods.

Here is the example that reproduces the problem.

```python
import pennylane as qml
from pennylane import numpy as np

import jax
import jax.numpy as jnp

@qml.qnode(dev)
def circuit_default(params):
    qml.templates.StronglyEntanglingLayers(params, wires=range(n_wires))
    return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1) @ qml.PauliX(2) @ qml.PauliY(3))


@qml.qnode(dev, diff_method="adjoint", interface="jax")
def circuit(params):
    qml.templates.StronglyEntanglingLayers(params, wires=range(n_wires))
    return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1) @ qml.PauliX(2) @ qml.PauliY(3))

rng = np.random.default_rng(seed=42)

params_shape = qml.templates.StronglyEntanglingLayers.shape(n_layers=2, n_wires=n_wires)
params1 = rng.standard_normal(params_shape)
params2 = rng.standard_normal(params_shape)
params1.requires_grad = True
params2.requires_grad = True

jp1 = jnp.array(params1)
jp2 = jnp.array(params2)

print(jax.grad(circuit)(jp1))
print(qml.grad(circuit_default)(params1))
```
```
jax:  [[[-0.0145612  -0.05081451  0.09873819]
  [-0.0145612  -0.1943438  -0.14168428]
  [-0.0145612  -0.22125678 -0.0504223 ]
  [-0.0145612  -0.08118758 -0.00099398]]

 [[ 0.00601515  0.04192177 -0.02662915]
  [-0.1740805  -0.09054054 -0.16497213]
  [-0.02582757 -0.51569396 -0.0145612 ]
  [-0.01456121 -0.01456121 -0.0145612 ]]]
default:  [[[ 1.21430643e-17 -3.62533011e-02  1.13299398e-01]
  [ 1.38777878e-17 -1.79782596e-01 -1.27123078e-01]
  [ 6.07153217e-18 -2.06695577e-01 -3.58610995e-02]
  [ 1.04083409e-17 -6.66263660e-02  1.35672224e-02]]

 [[ 2.05763547e-02  5.64829768e-02 -1.20679465e-02]
  [-1.59519298e-01 -7.59793138e-02 -1.50410913e-01]
  [-1.12663690e-02 -5.01132755e-01 -2.34187669e-17]
  [-2.08166817e-17  2.77555756e-17  6.07153217e-18]]]
```

**Comment by albi3ro** [comment]
Upon further investigation, the problem arises [here](https://github.com/PennyLaneAI/pennylane/blob/411c3d52cd87f612a1378467a6049d6b5fbdf17c/pennylane/interfaces/jax.py#L106), where the interface makes a copy of the tape.  The `tape.jacobian_options` does not get duplicated in the copy.

First question: why does the tape need to be copied? @Thenerdstation 

Second question: should [tape.copy](https://github.com/PennyLaneAI/pennylane/blob/411c3d52cd87f612a1378467a6049d6b5fbdf17c/pennylane/tape/tape.py#L1178) also duplicated the `jacobian_options` attribute?

**Comment by josh146** [comment]
Good catch @albi3ro. `tape.copy` is defined here: https://github.com/PennyLaneAI/pennylane/blob/master/pennylane/tape/tape.py#L1178. So a quick fix would probably be adding the following to `JacobianTape`:

```python
def copy(self, copy_operations=False, tape_cls=None):
    copied_tape = super().copy(copy_operations=copy_operations, tape_cls=tape_cls)
    copied_tape.jacobian_options = self.jacobian_options
    return copied_tape
```

Long term, an approach where we could avoid copying altogether would be nice.

**Comment by albi3ro** [comment]
Finite diff also fails with the jax interface, even after I have gotten the adjoint method to work.  Investigating that now.

**Comment by josh146** [comment]
Nice, definitely shows there are quite major holes in our test suite still 🤦

**Comment by albi3ro** [comment]
Float64 has to be manually enable for JAX, and float32 is too imprecise for finite differences. I'm writing in a `UserWarning` if the user tries to use finite-diff and jax together without enabling float64 support.

**Comment by albi3ro** [comment]
Closes with #1349  merge.
